sklearn/feature_extraction/text.py

```diff
@@ -1157,6 +1163,13 @@ def idf_(self):
         # which means hasattr(self, "idf_") is False
         return np.ravel(self._idf_diag.sum(axis=0))
 
+    @idf_.setter
+    def idf_(self, value):
+        value = np.asarray(value, dtype=np.float64)
+        n_features = value.shape[0]
```

Comment:
`n_features` is the vocabulary size, we should check here that the provided shape is consistent with it.

Reply:
TfidfTransformer  does not have the vocabulary, TfidfVectorizer does.  I added validation to TfidfVectorizer

Reply:
Yes, right, thanks.